### PR TITLE
build: make the ci work again

### PR DESCRIPTION
**Pull request by glerchundi**
While this project didn't receive many updates, the Go and Protobuf
community were evolving and they decided to move the development from
the being under to Go's umbrella to be on the Protobuf's. They also
decided to make the original project in github.com/golang/protobuf just
a wrapper around the new project located in
github.com/protocolbuffers/protobuf-go. This made it possible to include
a lot of fixed but at the same time a lot of breaking changes were
included.

Pinning to a version before doing this major bump on the Golang Protobuf
API (they coined as APIv2) solved the issue.

More info.: https://blog.golang.org/protobuf-apiv2

### CODE REVIEW

**Comment by glerchundi** [comment]
Reviewable @dkunitsk @akonradi!

**Comment by akonradi** [comment]
Fantastic, thank you!

**Review by akonradi — APPROVED** [review]

**Comment by akonradi** [comment]
@dkunitsk can you review? I don't have write access.

**Comment by dkunitsk** [comment]
I believe the assessment in the description is not quite correct. It's true that there exists a v2 API, but that is what's happening here. 

I compared the last successful CI run and a current failing one. github.com/golang/protobuf changed from v1.4.3 to v1.5.1 (and yes, this also pulled in a change to google.golang.org/protobuf, from 1.25.0 to 1.26.0. This is the v2 package, and the v1 package has been using it internally for a while). 

As I understand it, the change basically does what 1.4.3 has been warning about, which is a stricter disposition towards `go_package`. This is from 1.4.3:

```
2021/03/04 15:31:11 WARNING: Deprecated use of 'go_package' option without a full import path in "entities.proto", please specify:
	option go_package = ".;BAD_pack__age_name_";
A future release of protoc-gen-go will require the import path be specified.
See https://developers.google.com/protocol-buffers/docs/reference/go-generated#package for more information.
```

In v1.5.1, protoc now fails:
```
protoc-gen-go: invalid Go import path "BAD.pack--age.name$" for "entities.proto"
The import path must contain at least one forward slash ('/') character.
See https://developers.google.com/protocol-buffers/docs/reference/go-generated#package for more information.
--go_out: protoc-gen-go: Plugin failed with status code 1.
```

IIUC, pgs was designed to work with the looser rules about go_package, and these testdata inputs are testing how well pgs handles liberal uses of go_package. 

The fix that's needed here is to remove these test inputs and remove the test assertions that check the old behavior. 

I estimate it's maybe half a day of work, and I have not been able to find this time so far. 
@glerchundi @sarthak40 if this is something you could take on that would be extremely appreciated.

**Comment by dkunitsk** [comment]
At the same time, pinning for now (with a TODO) is better than the current state. But this would be quite urgent to fix. The failure we saw is actually the reason protoc-gen-go is not pinned - so that backward incompatibilities are caught in testing. So I am hesitant to pull the trigger on this. It's extremely improbable anyone would care to fix this in the future.

**Comment by sarthak40** [comment]
@dkunitsk if I understand this correctly, this the same fix proposed by @glerchundi in #85?

**Comment by glerchundi** [comment]
If I'm not mistaken this is what happened in the Protobuf & Go world since it was conceived:
1. Golang team created the initial development in `github.com/golang/protobuf`
2. They decided to move the development under the `github.com/protocolbuffers` organisation umbrella
3. Once under that umbrella they decided another two things:
    1. Make breaking changes and guarantee backward compatibility on a best effort way
    2. Wrap the old repo in `golang/protobuf` by using the `protocolbuffers/protobuf-go`

So the situation of the versioning is as follows:
* `golang/protobuf` <= v1.3.X has an APIv1 contracts & APIv1 implementation
* `golang/protobuf` >= 1.4 has an APIv1 contracts & an underlying implementation of APIv2
*  `protocolbuffers/protobuf-go` APIv2 contracts & implementation.

What I was doing here is just making the CI work back again. The reason behind this is that the tests that are currently implemented are assuming that the generator has an APIv1 implementation. That's why it those protect users from things that aren't happening anymore like packages with weird forms & `go_package`s without slashes. Because the APIv2 implementation already protects from generating code which is not valid.

So the proposal I made in https://github.com/lyft/protoc-gen-star/pull/85#issuecomment-829150282 was:
1. Not mandatory but merge this as is as this will fix the CI. It will just make it have a green light here but **the inconsistency keeps existing**! I mean, the `go.mod` already includes a version of `golang/protobuf` (1.4.2) which is implementing an APIv2  but the tests were expecting to have an APIv1 implementation.
2. Use the very same version for the library as well as for the tests, that is, the version >= 1.4. I'll deal with this one as soon as possible.
3. Bump the whole repo to start using the new implementation present in `protocolbuffers/protobuf-go`. Basically what @akonradi is doing in #86.

It's pretty clear that the end of the road is 3. but it will probably take more time to review & merge than the second option. This middle point would just lower the pressure on this repo as it's the base for other projects (e.g. `protoc-gen-validate`) that I don't know how prepared they would be start using the new APIv2 interfaces.

Sorry for the looong answer!

**Comment by dkunitsk** [comment]
@glerchundi thank you for the PR https://github.com/lyft/protoc-gen-star/pull/93. I will review it today. 

I believe your explanation above is generally correct but it has no relationship to the breakage we're experiencing.

**Comment by glerchundi** [comment]
Thanks for the prompt response @dkunitsk.

You're right that the last build that succeeded was a 1.4.x version! But I also think that what I explained still applies with an additional note. Reading the golang/protobuf release notes for 1.4, it seems like they applied *some* of those breaking changes but not all. In those that weren't migrated yet they added deprecation notes (those appear on the travis logs).

So the last build that succeeded was on **4th March**: Newest version of golang/protobuf on that date was **v1.4.3**.
The build that failed was run on **19th March**: Newest version of golang/protobuf on that date was **v1.5.1**.

Does it makes sense?

**Pull request closed without merging**
